superset/views/base.py

```diff
@@ -361,6 +371,13 @@ def show_superset_errors(ex: SupersetErrorsException) -> FlaskResponse:
     return json_errors_response(errors=ex.errors, status=ex.status)
 
 
+# Redirect to login if the CSRF token is expired
+@superset_app.errorhandler(CSRFError)
+def refresh_csrf_token(ex: CSRFError) -> FlaskResponse:
+    logger.warning(ex)
+    return redirect(appbuilder.get_url_for_login)
```

Comment:
Does this still make sense when the request is json? Might be kind of confusing to return an html redirect for a json request.

Reply:
Yeah, I agree. But do we currently have a mechanism to authenticate programmatically? Is there a better endpoint to redirect JSON requests?

Reply:
Yes take a look at the api docs, https://superset.apache.org/docs/rest-api

there’s a `security/login` route as well as a csrf route (though it looks like the docs are not updated with that one, should be available locally at `/swagger/v1/`)

Reply:
Cool, let me update it. Thanks!

Reply:
@nytai looking at the docs we can't redirect a JSON request there because it requires a `POST` payload with credentials.

I changed the method so that for API calls we show the SIP-40 error payload, and for non-API calls we redirect the user to `/login`.

Reply:
yea that makes sense, how about instead of checking the path we check `request.is_json`? https://tedboy.github.io/flask/generated/generated/flask.Request.is_json.html#flask-request-is-json

Reply:
Will do, thanks!